xtask/src/release.rs

```diff
@@ -1,67 +1,147 @@
-use std::path::Path;
-
+use std::{
+    io::{stdin, stdout, BufRead, Write},
+    path::{Path, PathBuf},
+};
+
+use isahc::{
+    auth::{Authentication, Credentials},
+    config::Configurable,
+    http::StatusCode,
+    HttpClient, ReadResponseExt, Request,
+};
 use itertools::Itertools;
+use semver::Version;
 use serde::Deserialize;
-use serde_json::json;
+use serde_json::{json, Value as JsonValue};
 use toml::from_str as from_toml_str;
-use xshell::{cmd, pushd, read_file};
+use xshell::{pushd, read_file};
 
-use crate::{config, flags, project_root, Result};
+use crate::{cmd, config, Result};
```

Comment:
I thought importing `macro_rules!` macros from the same crate with `use` wasn't a thing yet.. Does this work back to Rust 1.45?

Reply:
As the checks passed this should work, right?

Reply:
Not necessarily. The MSRV checks are limited and don't cover xtask at all I think. It's probably fine either way, just curious.

Reply:
Not necessarily. The MSRV checks are limited and don't cover xtask at all I think. It's probably fine either way, just curious.